src/test/java/teammates/storage/api/NotificationsDbTest.java

```diff
@@ -0,0 +1,359 @@
+    private final Map<String, NotificationAttributes> typicalNotifications = getTypicalDataBundle().notifications;
+
+    // Notification attributes will be assigned with value at setup()
+    private NotificationAttributes n;
+    private Set<String> expectedIds = new HashSet<>();
+
+    @BeforeMethod
+    public void setup(Method method) throws Exception {
+        switch (method.getName()) {
+        case "testGetNotification":
+        case "testUpdateNotification":
+        case "testUpdateNotification_bulkUpdate_shouldUpdateSuccessfully":
+        case "testUpdateNotification_singleFieldUpdate_shouldUpdateSuccessfully":
+        case "testDeleteNotification":
+            n = notificationsDb.createEntity(typicalNotifications.get("notification1"));
+            break;
+        case "testGetAllNotifications":
+            for (NotificationAttributes n : typicalNotifications.values()) {
+                notificationsDb.createEntity(n);
+                expectedIds.add(n.getNotificationId());
+            }
+            break;
+        case "testGetActiveNotificationsByTargetUser":
+        case "testCreateNotification":
+        case "testHasExistingEntities":
+            // Do nothing - leave the initialization to the test method.
+            break;
+        default:
+            // All test methods should be covered above, so this should not be reached.
+            assert false;
+            break;
+        }
+    }
+
+    /**
+     * Removes all notifications created by each test.
+     */
+    @AfterMethod
+    public void cleanUp() {
+        List<NotificationAttributes> retrieved = notificationsDb.getAllNotifications();
+        assertNotNull(retrieved);
+
+        ______TS("clean up: delete " + retrieved.size() + " notifications");
+        retrieved.forEach(n -> notificationsDb.deleteNotification(n.getNotificationId()));
+
+        n = null;
+        expectedIds.clear();
+    }
+
+    @Test
+    public void testGetNotification() throws Exception {
+        ______TS("typical success case");
+        NotificationAttributes retrieved = notificationsDb.getNotification(n.getNotificationId());
+        assertNotNull(retrieved);
+
+        ______TS("expect null for non-existent account");
+        retrieved = notificationsDb.getNotification(NON_EXISTENT_ID);
+        assertNull(retrieved);
+
+        ______TS("failure: null parameter");
+        assertThrows(AssertionError.class, () -> notificationsDb.getNotification(null));
+    }
+
+    @Test
+    public void testGetAllNotifications() throws Exception {
+        ______TS("typical success case");
+        List<NotificationAttributes> retrieved = notificationsDb.getAllNotifications();
+
+        // Verifies that what was retrieved is exactly the same as what was created.
+        this.verifyNotifications(retrieved, expectedIds);
+    }
+
+    @Test
+    public void testGetActiveNotificationsByTargetUser() throws Exception {
```

Comment:
Is there any way to simplify this test? If we consider the opposite, we are just trying to test that only notifications which have started, not expired, has the intended target user should be returned, why not create some active and non-active notifications in `typicalDataBundle.json` based on these requirements and verify that only those will be returned.